phpunit/html/wp-html-walker-test.php

```diff
@@ -0,0 +1,1012 @@
+			  Squash and merge
+			</button>
+
+			<button type="button" class="merge-box-button btn-group-rebase rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Rebase and merge
+			</button>
+
+			<button aria-label="Select merge method" disabled="disabled" type="button" data-view-component="true" class="select-menu-button btn BtnGroup-item"></button>
+		</div>
+	</div>
+</div>
+HTML;
+
+		$expected_output = <<<HTML
+<div data-details="{ &quot;key&quot;: &quot;value&quot; }" selected class="merge-message is-processed" checked>
+	<div class="select-menu d-inline-block">
+		<div checked class=" MixedCaseHTML position-relative button-group Another-Mixed-Case" />
+		<div checked class=" MixedCaseHTML position-relative button-group Another-Mixed-Case">
+			<button type="button" class="merge-box-button btn-group-merge rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Merge pull request
+			</button>
+
+			<button type="button" class="merge-box-button btn-group-squash rounded-left-2 btn  BtnGroup-item js-details-target hx_create-pr-button" aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Squash and merge
+			</button>
+
+			<button type="button"  aria-expanded="false" data-details-container=".js-merge-pr" disabled="">
+			  Rebase and merge
+			</button>
+
+			<button aria-label="Select merge method" disabled="disabled" type="button" data-view-component="true" class="select-menu-button btn BtnGroup-item"></button>
+		</div>
+	</div>
+</div>
+HTML;
+
+		$w = new WP_HTML_Walker( $input );
+		$this->assertTrue( $w->next_tag( 'div' ) );
+		$w->set_attribute( 'data-details', '{ "key": "value" }' );
+		$w->add_class( 'is-processed' );
+		$this->assertTrue(
+			$w->next_tag(
+				array(
+					'tag_name'   => 'div',
+					'class_name' => 'BtnGroup',
+				)
+			)
+		);
+		$w->remove_class( 'BtnGroup' );
+		$w->add_class( 'button-group' );
+		$w->add_class( 'Another-Mixed-Case' );
+		$this->assertTrue(
+			$w->next_tag(
+				array(
+					'tag_name'   => 'div',
+					'class_name' => 'BtnGroup',
+				)
+			)
+		);
+		$w->remove_class( 'BtnGroup' );
+		$w->add_class( 'button-group' );
+		$w->add_class( 'Another-Mixed-Case' );
+		$this->assertTrue(
+			$w->next_tag(
+				array(
+					'tag_name'     => 'button',
+					'class_name'   => 'btn',
+					'match_offset' => 3,
+				)
+			)
+		);
+		$w->remove_attribute( 'class' );
+		$this->assertFalse( $w->next_tag( 'non-existent' ) );
+		$w->set_attribute( 'class', 'test' );
+		$this->assertSame( $expected_output, (string) $w );
+	}
+
+	/**
+	 * @ticket 56299
+	 */
+	public function test_works_with_single_quote_marks() {
```

Comment:
The test name should indicate what "works with single quote marks".

Similar point for other test methods.

Reply:
I updated this specific method in 19a85460eb

Reply:
I updated all other methods in c96f08bd76